api/tests/tests.py

```diff
@@ -0,0 +1,75 @@
+import json
+from django.test import TestCase, Client
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+
+from posts.models import Post, ContentType
+
+from .constants import POST_DATA, POST_IMG_DATA
+
+
+class AuthorTests(TestCase):
+    def setUp(self) -> None:
+        self.client = Client()
+        get_user_model().objects.create_user(username='bob', password='password')
+
+    def test_authors(self):
+        self.client.login(username='bob', password='password')
+        res = self.client.get('/api/v1/authors/')
+        self.assertEqual(res.status_code, 200)
+        body = json.loads(res.content.decode('utf-8'))
+        self.assertEqual(body['type'], 'authors')
+        self.assertEqual(len(body['items']), 1)
+
+    def test_authors_require_login(self):
+        res = self.client.get('/api/v1/authors/')
+        self.assertEqual(res.status_code, 403)
+
+    def test_create_author(self):
+        self.client.login(username='bob', password='password')
+        res = self.client.post('/api/v1/authors/', {'username': 'alice', 'password': 'password'})
+        self.assertEqual(res.status_code, 405)
+
+
+class ImageTests(TestCase):
+    def setUp(self) -> None:
+        self.client = Client()
+        self.author = get_user_model().objects.create_user(username='bob', password='password')
+
+        self.post = Post.objects.create(
+            title=POST_DATA['title'],
+            description=POST_DATA['description'],
+            content_type=POST_DATA['content_type'],
+            content=POST_DATA['content'],
+            author_id=self.author.id,
+            unlisted=POST_DATA['unlisted'])
+        self.post.full_clean()
```

Comment:
> Note that full_clean() will not be called automatically when you call your model’s [save()](https://docs.djangoproject.com/en/4.0/ref/models/instances/#django.db.models.Model.save) method. You’ll need to call it manually when you want to run one-step model validation for your own manually created models. For example:

[src](https://docs.djangoproject.com/en/4.0/ref/models/instances/#validating-objects-1)

I did not know this, thank you for validating